lib/libc/Kconfig

```diff
@@ -55,6 +55,18 @@ config HAS_NEWLIB_LIBC_NANO
 
 if NEWLIB_LIBC
 
+config ZEPHYR_TOOLCHAIN_VARIANT
+	string
+	option env="ZEPHYR_TOOLCHAIN_VARIANT"
+
+config TOOLCHAIN_HAS_NEWLIB_LIBC_NANO
+	bool "Toolchain has newlib-nano"
+	depends on ZEPHYR_TOOLCHAIN_VARIANT = "cross-compile"
+	select HAS_NEWLIB_LIBC_NANO
+	help
+	  Select if "cross-compile" toolchain has nano version of
+	  Newlib C library
```

Comment:
There are two major issues with this approach:

1. This is essentially the same thing as HAS_NEWLIB_LIBC_NANO in which case we should just add prompt to HAS_NEWLIB_LIBC_NANO instead of introducing another configuration -- but that is not the correct thing to do as I will explain later.
2. If this is intended to be solely for cross-compile, it should not belong in the libc config; rather, it should be part of the toolchain config.